group-coordinator/src/test/java/org/apache/kafka/coordinator/group/streams/topics/CopartitionedTopicsEnforcerTest.java

```diff
@@ -0,0 +1,261 @@
+        );
+        final CopartitionedTopicsEnforcer enforcer =
+            new CopartitionedTopicsEnforcer(LOG_CONTEXT, topicPartitionProvider(topicPartitionCounts));
+
+        final Map<String, Integer> result =
+            enforcer.enforce(
+                Set.of(firstSourceTopic, secondSourceTopic, repartitionTopic),
+                Set.of(),
+                Set.of(repartitionTopic)
+            );
+
+        assertEquals(Map.of(repartitionTopic, 2), result);
+    }
+
+
+    @Test
+    public void shouldSetNumPartitionsToMaximumPartitionsWhenAllTopicsAreRepartitionTopics() {
+        final String repartitionTopic1 = "repartitionTopic1";
+        final String repartitionTopic2 = "repartitionTopic2";
+        final String repartitionTopic3 = "repartitionTopic3";
+        final Map<String, Integer> topicPartitionCounts = Map.of(
+            repartitionTopic1, 1,
+            repartitionTopic2, 15,
+            repartitionTopic3, 5
+        );
+        final CopartitionedTopicsEnforcer enforcer =
+            new CopartitionedTopicsEnforcer(LOG_CONTEXT, topicPartitionProvider(topicPartitionCounts));
+
+        final Map<String, Integer> result = enforcer.enforce(
+            Set.of(repartitionTopic1, repartitionTopic2, repartitionTopic3),
+            Set.of(),
+            Set.of(repartitionTopic1, repartitionTopic2, repartitionTopic3)
+        );
+
+        assertEquals(Map.of(
+            repartitionTopic1, 15,
+            repartitionTopic2, 15,
+            repartitionTopic3, 15
+        ), result);
+    }
+
+    @Test
+    public void shouldThrowAnExceptionIfTopicInfosWithEnforcedNumOfPartitionsHaveDifferentNumOfPartitions() {
+        final String repartitionTopic1 = "repartitioned-1";
+        final String repartitionTopic2 = "repartitioned-2";
+        final Map<String, Integer> topicPartitionCounts = Map.of(
+            repartitionTopic1, 10,
+            repartitionTopic2, 5
+        );
+        final CopartitionedTopicsEnforcer enforcer =
+            new CopartitionedTopicsEnforcer(LOG_CONTEXT, topicPartitionProvider(topicPartitionCounts));
+
+        final TopicConfigurationException ex = assertThrows(
+            TopicConfigurationException.class,
+            () -> enforcer.enforce(
+                Set.of(repartitionTopic1, repartitionTopic2),
+                Set.of(repartitionTopic1, repartitionTopic2),
+                Set.of()
+            )
+        );
+
+        final TreeMap<String, Integer> sorted = new TreeMap<>(
+            Utils.mkMap(Utils.mkEntry(repartitionTopic1, 10),
+                Utils.mkEntry(repartitionTopic2, 5))
+        );
+        assertEquals(Status.INCORRECTLY_PARTITIONED_TOPICS, ex.status());
+        assertEquals(String.format(
+            "Following topics do not have the same number of partitions: " +
+                "[%s]", sorted), ex.getMessage());
+    }
+
+    @Test
+    public void shouldNotThrowAnExceptionWhenTopicInfosWithEnforcedNumOfPartitionsAreValid() {
```

Comment:
nit:
Could you try to formulate this in a positive way?

Reply:
Done